modules/fedimint-mint-tests/src/bin/mint-client-sanity.rs

```diff
@@ -1,80 +1,4 @@
-use anyhow::Context;
-use devimint::cmd;
-use tracing::info;
-
 #[tokio::main]
 async fn main() -> anyhow::Result<()> {
-    devimint::run_devfed_test(|fed, _process_mgr| async move {
-        let fed = fed.fed().await?;
-
-        test_note_consoliation(fed).await?;
-        Ok(())
-    })
-    .await
-}
-
-/// Test note consolidation, which at the time of writing basically means that
-/// once client accumulates too many notes of certain denomination, any
-/// transaction building will include excessive notes as extra inputs, to
-/// consolidate them into higher denominations.
-///
-/// In the future we will probably change the whole thing and delete this thing.
-async fn test_note_consoliation(fed: &devimint::federation::Federation) -> anyhow::Result<()> {
-    let sender = fed.new_joined_client("sender").await?;
-    let receiver = fed.new_joined_client("receiver").await?;
-
-    let can_no_wait = cmd!(sender, "reissue", "--help")
-        .out_string()
-        .await?
-        .contains("no-wait");
-
-    if !can_no_wait {
-        info!("Version before `--no-wait` didn't have consolidation implemented");
-        return Ok(());
-    }
-    fed.pegin_client(10_000, &sender).await?;
-
-    let mut all_notes = vec![];
-    for i in 0..20 {
-        let info = cmd!(sender, "info").out_json().await?;
-        info!(%info, "sender info");
-        // remint sender notes from time to time to make sure it have 1msat notes
-        if i % 2 == 1 {
-            let notes = cmd!(sender, "spend", "1sat",).out_json().await?["notes"]
-                .as_str()
-                .context("invoice must be string")?
-                .to_owned();
-
-            cmd!(sender, "reissue", notes).run().await?;
-        }
-
-        let notes = cmd!(sender, "spend", "1msat",).out_json().await?["notes"]
-            .as_str()
-            .context("invoice must be string")?
-            .to_owned();
-
-        all_notes.push(notes);
-    }
-
-    for notes in &all_notes[..all_notes.len() - 1] {
-        cmd!(receiver, "reissue", "--no-wait", notes).run().await?;
-    }
-
-    // wait for all at the same time to make things go faster
-    cmd!(receiver, "dev", "wait-complete").run().await?;
-
-    // reissuance of last note will trigger consolidation
-    cmd!(receiver, "reissue")
-        .args(&all_notes[all_notes.len() - 1..])
-        .run()
-        .await?;
-
-    let info = cmd!(receiver, "info").out_json().await?;
-    info!(%info, "receiver info");
-    // receiver has the balance
-    assert_eq!(info["total_amount_msat"].as_i64().unwrap(), 20);
```

Comment:
I'm guessing not comparing the balance, and just asserting that there is fewer than 20 notes, should make it pass? @joschisan

Reply:
Probably yes but not for the right reason I think. I you reissue a 1msat note by itself with fees its unpredictable what kind of output denominations you will get.

Reply:
That's why I removed the test as it seems to rely on a transaction being finalized with specific denominations.

Reply:
Could we migrate this test to a different run of devimint where the fees are set to 0?

Reply:
Or rather: if `run_devfed_test` accepted arguments that allow setting fees to `0` then we could test stuff that requires no fees with no fees.